serving/samples/helloworld-python/Dockerfile

```diff
@@ -9,10 +9,12 @@ COPY . .
 
 # Install production dependencies.
 RUN pip install Flask
+RUN pip install gunicorn
 
 # Service must listen to $PORT environment variable.
 # This default value facilitates local development.
 ENV PORT 8080
 
-# Run the web service on container startup.
-CMD ["python", "app.py"]
+# Run the web service on container startup. Here we use the gunicorn
+# webserver, with one worker process and 8 threads.
+CMD exec gunicorn -b :$PORT -w 1 -t 8 app:app
```

Comment:
Given the guidance from the [gunicorn settings page](http://docs.gunicorn.org/en/stable/settings.html#workers) of 2-4 x $(NUM_CORES), could we add a comment to that effect? Should we default to 2 on the assumption of at least one core and more concurrency by default? Knative has a [default concurrency of 100](https://github.com/knative/serving/blob/3f00c39e289ed4bfb84019131651c2e4ea660ab5/config/config-autoscaler.yaml#L35)

We could go fancy but it seems excessive here: `expr 2 \* $(nproc)`, we could try to provide that tip in the README if you think it's worthwhile.

Reply:
We are using threads (-t 8) here, so we only need one worker per core. That advice is out of date. Unfortunately the Run environment currently doesn't have a concrete idea of cores; CPU is allocated based on virtual CPU time and not on cores. Since we can't reliably read the cores from cpuinfo or even expect the customer to know how many cores are available, it's a hard problem to solve. I'm fine with including guidance in comments but couldn't figure out a way to summarize it elegantly.

Reply:
Whoops, it turns out I needed --threads=8 for that to work. Changed as per your suggestion to use longform versions.